internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -2046,4 +2048,485 @@ ExecExprVisitor::visit(ExistsExpr& expr) {
+compareTwoJsonArray(
+    simdjson::simdjson_result<simdjson::fallback::ondemand::array> arr1,
+    const proto::plan::Array& arr2) {
+    if (arr2.array_size() != arr1.count_elements()) {
+        return false;
+    }
+    int i = 0;
+    for (auto&& it : arr1) {
+        //    for (int i = 0; i < arr2.array_size(); i++) {
+        switch (arr2.array(i).val_case()) {
+            case proto::plan::GenericValue::kBoolVal: {
+                auto val = it.template get<bool>();
+                if (val.error()) {
+                    return false;
+                }
+                if (val.value() != arr2.array(i).bool_val()) {
+                    return false;
+                }
+                break;
+            }
+            case proto::plan::GenericValue::kInt64Val: {
+                auto val = it.template get<int64_t>();
+                if (val.error()) {
+                    return false;
+                }
+                auto x = val.value();
+                if (x != arr2.array(i).int64_val()) {
+                    return false;
+                }
+                break;
+            }
+            case proto::plan::GenericValue::kFloatVal: {
+                auto val = it.template get<double>();
+                if (val.error()) {
+                    return false;
+                }
+                if (val.value() != arr2.array(i).float_val()) {
+                    return false;
+                }
+                break;
+            }
+            case proto::plan::GenericValue::kStringVal: {
+                auto val = it.template get<std::string_view>();
+                if (val.error()) {
+                    return false;
+                }
+                if (val.value() != arr2.array(i).string_val()) {
+                    return false;
+                }
+                break;
+            }
+            default:
+                PanicInfo(fmt::format("unsupported data type {}",
+                                      arr2.array(i).val_case()));
+        }
+        i++;
+    }
+    return true;
+}
+
+template <typename ExprValueType>
+auto
+ExecExprVisitor::ExecJsonContains(JsonContainsExpr& expr_raw) -> BitsetType {
+    using Index = index::ScalarIndex<milvus::Json>;
+    auto& expr = static_cast<JsonContainsExprImpl<ExprValueType>&>(expr_raw);
+    auto pointer = milvus::Json::pointer(expr.column_.nested_path);
+    auto index_func = [](Index* index) { return TargetBitmap{}; };
+    auto elements = expr.elements_;
+    auto elem_func = [&elements, &pointer](const milvus::Json& json) {
+        using GetType =
+            std::conditional_t<std::is_same_v<ExprValueType, std::string>,
+                               std::string_view,
+                               ExprValueType>;
+        auto doc = json.doc();
+        auto array = doc.at_pointer(pointer).get_array();
+        if (array.error()) {
+            return false;
+        }
+        for (auto&& it : array) {
+            for (auto element : elements) {
```

Comment:
Also this algorithm is not good enough, fix $n$ is the length of array, and $m$ the length of elements.
This performs $O(nm)$, you could optimize this by converting the elements to a `unordered_set`, and then lookup it, then it will boost to $O(n)$